include/ringmesh/basic/types.h

```diff
@@ -96,12 +96,33 @@ namespace RINGMesh
         return true;
     }
 
+    template < index_t DIMENSION >
+    bool operator==(
+        const sivecn< DIMENSION >& u, const sivecn< DIMENSION >& v )
+    {
+        for( index_t i = 0; i < DIMENSION; i++ )
```

Comment:
You could use range here.

Reply:
actually, no, because range is defined in range.h which uses definitions from types.h, and it left some issues if I did (I had to do the same kind of operator for vecn and since it brought this issue we decided to leave "conventional" for-loops in types.h if I remember correctly)